build/gen_stub.php

```diff
@@ -1332,6 +1390,288 @@ private function getFlagsAsString(): string
 
         return implode("|", $flags);
     }
+
+    /**
+     * @param ClassInfo[] $classMap
+     */
+    public function getClassSynopsisDocument(array $classMap): ?string {
+
+        $doc = new DOMDocument();
+        $doc->formatOutput = true;
+        $classSynopsis = $this->getClassSynopsisElement($doc, $classMap);
+        if (!$classSynopsis) {
+            return null;
+        }
+
+        $doc->appendChild($classSynopsis);
+
+        return $doc->saveXML();
```

Comment:
Might consider to preprend 3 spaces here.

Reply:
Yeah, I wanted to do so, but I wasn't able to add spaces before the root element - at least via ext/dom. To be fair, `str_replace` would probably work. ^^